src/http.rs

```diff
@@ -12,44 +14,54 @@ pub fn retrieve_asset(
     opt_insecure: bool,
 ) -> Result<(String, String), reqwest::Error> {
     if is_data_url(&url).unwrap() {
+        eprintln!("is data URL [ {} ]", &url);
         Ok((url.to_string(), url.to_string()))
     } else {
-        let client = Client::builder()
-            .timeout(Duration::from_secs(10))
-            .danger_accept_invalid_certs(opt_insecure)
-            .build()?;
-        let mut response = client.get(url).header(USER_AGENT, opt_user_agent).send()?;
+        if cache.contains_key(&url.to_string()) {
+            // url is in cache
+            let data = cache.get(&url.to_string()).unwrap();
+            Ok((data.to_string(), url.to_string()))
+        } else {
+            // url not in cache, we request it
+            let client = Client::builder()
+                .timeout(Duration::from_secs(10))
+                .danger_accept_invalid_certs(opt_insecure)
+                .build()?;
+            let mut response = client.get(url).header(USER_AGENT, opt_user_agent).send()?;
 
-        if !opt_silent {
-            if url == response.url().as_str() {
-                eprintln!("[ {} ]", &url);
-            } else {
-                eprintln!("[ {} -> {} ]", &url, &response.url().as_str());
+            if !opt_silent {
+                if url == response.url().as_str() {
+                    eprintln!("[ {} ]", &url);
```

Comment:
I'm thinking that we could copy this line and put it right after line 22, e.g. something like:

     eprintln!("[ {} ] (from cache)", &url);

this way we could indicate that those resources were retrieved, but came from cache (I'm thinking of adding a -C flag to disable cache in the future in case caching breaks anything)

Reply:
do you want this to be wrapped in `if !opt_silent {` ? so it can be suppressed ?

Reply:
`if !opt_silent {` would make sense, good catch!